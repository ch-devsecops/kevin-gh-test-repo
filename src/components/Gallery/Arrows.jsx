import React from 'react';
import keyCodes from '@honda-canada/js-utilities/lib/keyCodes';
import keypressCallback from '@honda-canada/js-utilities/lib/keypressCallback';
import styled from 'styled-components';
import css from '@styled-system/css';
import { Box, Icon, IconWrapper, useThemeContext } from '@honda-canada/design-system-react';
import { ACURA_THEME_NAME } from '../../utils/constants';
import { getGtmTagValue } from '../../utils/gtmEvents';

export const ArrowContainer = ({ children, ...props }) => (
  <Box position="absolute" height="100%" top="0" display="flex" alignItems="center" {...props}>
    {children}
  </Box>
);

const StyledIconWrapper = styled(IconWrapper)(({ theme }) =>
  css({
    '&:focus-visible': {
      outline: `solid 8px ${theme.colors.aodaFocused}`,
    },
  }),
);

export const ArrowButton = ({ onClick, name, disabled = false, gtmTags, isDark, ...otherProps }) => {
  const { name: themeName, colors } = useThemeContext();
  const getIconColor = () => {
    if (isDark) {
      return disabled ? colors.grey[1] : 'white';
    }
    if (themeName === ACURA_THEME_NAME) {
      return disabled ? colors.grey[3] : colors.grey[0];
    }
    return disabled ? colors.grey[2] : colors.black;
  };

  return (
    <StyledIconWrapper
      onClick={onClick}
      onKeyDown={keypressCallback(keyCodes.ENTER, onClick)}
      as="button"
      backgroundColor={isDark ? 'black' : 'white'}
      border="none"
      cursor={!disabled ? 'pointer' : 'initial'}
      disabled={disabled}
      {...otherProps}
      data-gtm-model={getGtmTagValue(gtmTags?.modelName)}
      data-gtm-body-style={getGtmTagValue(gtmTags?.bodyStyle)}
      data-gtm-interaction-type={getGtmTagValue(gtmTags?.interactionType)}
      data-gtm-title={getGtmTagValue(gtmTags?.title)}
    >
      <Icon name={name} iconSize={['default', 'large']} color={getIconColor()} />
    </StyledIconWrapper>
  );
};
