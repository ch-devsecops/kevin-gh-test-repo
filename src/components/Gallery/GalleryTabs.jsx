import React from 'react';
import PropTypes from 'prop-types';
import { Column, Copy, Row } from '@honda-canada/design-system-react';
import themeStyles from './Gallery.styles';
import { getGtmTagValue } from '../../utils/gtmEvents';

const GalleryTab = themeStyles.apply(Column, 'GalleryTab');
const GalleryTabTitle = themeStyles.apply(Copy, 'GalleryTabTitle');

const GalleryTabs = ({ tabs = [], setActiveTab, activeTab, isDark, isFullScreenMobile }) => (
  <Row pl={['m', 'm', 'zero']} backgroundColor={isDark && !isFullScreenMobile ? 'black' : 'white'}>
    {tabs?.map((tab, i) =>
      tab.title ? (
        <GalleryTab
          as="button"
          key={i.toString()}
          isActive={activeTab === i}
          onClick={() => setActiveTab(i)}
          data-gtm-title={getGtmTagValue(tab?.gtmTitle)}
          data-gtm-model={getGtmTagValue(tab?.modelName)}
          data-gtm-body-style={getGtmTagValue(tab?.bodyStyle)}
          data-gtm-interaction-type={getGtmTagValue(tab?.interactionType)}
        >
          <GalleryTabTitle size="small" color={isDark && !isFullScreenMobile ? 'white' : 'black'}>
            {tab.title}
          </GalleryTabTitle>
        </GalleryTab>
      ) : null,
    )}
  </Row>
);

GalleryTabs.propTypes = {
  tabs: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string,
      gtmTitle: PropTypes.string,
      modelName: PropTypes.string,
      bodyStyle: PropTypes.string,
      interactionType: PropTypes.string,
    }),
  ),
  setActiveTab: PropTypes.func,
  activeTab: PropTypes.number,
  isDark: PropTypes.bool,
};

export default GalleryTabs;
