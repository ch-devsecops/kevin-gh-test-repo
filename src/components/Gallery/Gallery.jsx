import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import {
  Box,
  Icon,
  IconWrapper,
  useCarouselControl,
  useMediaQueries,
  Wrapper,
} from '@honda-canada/design-system-react';
import GalleryCarousel from './GalleryCarousel';
import GalleryTabs from './GalleryTabs';
import Thumbnails from './Thumbnails';
import ThumbnailControls from './ThumbnailControls';
import FullScreenMode from './FullScreenMode/FullScreenMode';
import themeStyles from './Gallery.styles';
import { getGtmTagValue } from '../../utils/gtmEvents';

const FullScreenButton = themeStyles.apply(IconWrapper, 'FullScreenButton');

const Gallery = ({ gallery = [], strings = {}, gtmTags = {}, isDark, isShort, ...otherProps }) => {
  const [activeCategoryIndex, setActiveCategoryIndex] = useState(0);

  const activeGallery = gallery[activeCategoryIndex]?.images; // images + content
  const activeImages = activeGallery?.map(item => item.image); // get only the images

  const { isDesktop, isMobile } = useMediaQueries();
  const [fullScreenMode, setFullScreenMode] = useState(false);
  const [currentPage, setPage] = useState(1); // thumbnail state
  const carouselControls = useCarouselControl(activeImages?.length, false);
  const isFullScreenMobile = fullScreenMode && isMobile;
  const { currentSlide, setCurrentSlide } = carouselControls;
  const activeContent = activeGallery[currentSlide]?.content;

  const thumbnailsNum = isDesktop ? 6 : 4;
  const thumbnailPages = Math.ceil(activeImages.length / thumbnailsNum) || 0;
  const thumbnailsId = `thumbnails-${thumbnailPages}`;
  const tabs =
    gallery?.map(category => ({
      title: gallery.length === 1 ? gallery[0].title : category.title,
      gtmTitle: gallery.length === 1 ? gallery[0].gtmTitle : category.gtmTitle,
      modelName: gtmTags.modelName,
      bodyStyle: gtmTags.bodyType,
      interactionType: 'gallery - image filter',
    })) || [];

  useEffect(() => {
    // when switching category, if current image is not found, reset current image
    if (currentSlide > activeImages.length - 1) {
      setCurrentSlide(0);
    }
  }, [activeCategoryIndex, activeImages.length]);

  useEffect(() => {
    // when switching category, if thumbnail page is not found, reset thumbnail page
    if (currentPage > thumbnailPages) setPage(1);
  }, [currentPage, thumbnailPages]);

  useEffect(() => {
    // when active carousel slide changes, check if it is within range of active thumbnail page
    const max = currentPage * thumbnailsNum;
    const min = max - thumbnailsNum + 1;

    const activeSlide = currentSlide + 1;
    const withinRange = activeSlide <= max && activeSlide >= min;

    if (!withinRange) {
      const updatedThumbnailPage = Math.ceil((currentSlide + 1) / thumbnailsNum);
      setPage(updatedThumbnailPage);
    }
  }, [currentSlide]);

  const GalleryTabsComponent = (
    <GalleryTabs
      tabs={tabs}
      setActiveTab={setActiveCategoryIndex}
      activeTab={activeCategoryIndex}
      isDark={isDark}
      isFullScreenMobile={isFullScreenMobile}
    />
  );

  const ThumbnailsComponent = (
    <Thumbnails
      images={activeImages}
      setActiveImageIndex={setCurrentSlide}
      activeImageIndex={currentSlide}
      currentPage={currentPage}
      thumbnailPages={thumbnailPages}
      id={thumbnailsId}
      isDark={isDark}
      isFullScreenMobile={isFullScreenMobile}
    />
  );

  const ThumbnailControlsComponent = (
    <ThumbnailControls
      pages={thumbnailPages}
      currentPage={currentPage}
      setPage={setPage}
      controlId={thumbnailsId}
      strings={strings}
      isDark={isDark}
      isFullScreenMobile={isFullScreenMobile}
    />
  );

  return (
    <Wrapper
      data-gtm-category={getGtmTagValue(gtmTags?.category)}
      data-gtm-component-type={getGtmTagValue(gtmTags?.type)}
      px="0 !important"
      {...otherProps}
      backgroundColor={isDark ? 'black' : 'white'}
    >
      {GalleryTabsComponent}
      <Box position="relative" height={isShort ? null : '100%'}>
        <FullScreenButton
          as="button"
          onClick={() => setFullScreenMode(true)}
          aria-label={strings.fullscreenModeAria}
          backgroundColor={isDark ? 'black' : 'white'}
        >
          <Icon name="expand" color={isDark ? 'white' : 'black'} />
        </FullScreenButton>
        <GalleryCarousel
          slides={activeImages}
          setFullScreenMode={setFullScreenMode}
          controls={carouselControls}
          gtmTags={{
            modelName: gtmTags.modelName,
            bodyStyle: gtmTags.bodyType,
            title: tabs[activeCategoryIndex]?.gtmTitle,
          }}
          strings={strings}
          height={['180px', '432px', '702px']}
          mobileLandscapeHt="280px"
          isDark={isDark}
          isShort={isShort}
        />
      </Box>

      {ThumbnailsComponent}
      {ThumbnailControlsComponent}

      <FullScreenMode
        onClose={() => setFullScreenMode(false)}
        isOpen={fullScreenMode}
        images={activeImages}
        content={activeContent}
        setActiveImageIndex={setCurrentSlide}
        controls={carouselControls}
        strings={strings}
        thumbnails={ThumbnailsComponent}
        thumbnailControls={ThumbnailControlsComponent}
        isDark={isDark}
        tabs={tabs}
        setActiveTab={setActiveCategoryIndex}
        activeTab={activeCategoryIndex}
        isFullScreenMobile={isFullScreenMobile}
      />
    </Wrapper>
  );
};

Gallery.propTypes = {
  /**
   * images and content
   */
  gallery: PropTypes.arrayOf(
    PropTypes.shape({
      image: PropTypes.element,
      content: PropTypes.shape({
        title: PropTypes.element,
        bodyText: PropTypes.element,
        ctas: PropTypes.element,
      }),
    }),
  ),
  /**
   * labels and aria labels
   */
  strings: PropTypes.shape({
    galleryModalAria: PropTypes.string,
    exitModalAria: PropTypes.string,
    fullscreenModeAria: PropTypes.string,
    nextImageAria: PropTypes.string,
    prevImageAria: PropTypes.string,
    nextThumbnailPageAria: PropTypes.string,
    prevThumbnailPageAria: PropTypes.string,
    hideAria: PropTypes.string,
    hideLabel: PropTypes.string,
    mobileLandscapeInfoLabel: PropTypes.string,
    learnMoreLabel: PropTypes.string,
    learnMoreAria: PropTypes.string,
    viewAllImagesLabel: PropTypes.string,
    viewAllImagesAria: PropTypes.string,
  }),
  gtmTags: PropTypes.shape({
    category: PropTypes.string,
    title: PropTypes.string,
    type: PropTypes.string,
    modelName: PropTypes.string,
    bodyType: PropTypes.string,
  }),
  isDark: PropTypes.bool,
  isShort: PropTypes.bool,
};

export default Gallery;
