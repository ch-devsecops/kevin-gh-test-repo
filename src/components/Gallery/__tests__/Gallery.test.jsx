import React from 'react';
import { jssRender, cleanup } from 'test-utils';
import Gallery from '../index';

window.scrollTo = jest.fn();

const mockData = require('./mockData.test.json');

afterEach(cleanup);

describe('<Gallery>', () => {
  test('should render correctly', () => {
    window.HTMLElement.prototype.scrollTo = () => {};
    const { container } = jssRender(<Gallery fields={mockData.fields} />);

    expect(container).toMatchSnapshot();
    expect(container.children[0]).toBeTruthy();
  });
});

describe('Gallery Short Variant', () => {
  test('should render short variant correctly', () => {
    window.HTMLElement.prototype.scrollTo = () => {};
    const { container } = jssRender(<Gallery fields={mockData.fields} params={{ styleType: 'Short' }} />);

    expect(container).toMatchSnapshot();
    expect(container.children[0]).toBeTruthy();
  });
});
