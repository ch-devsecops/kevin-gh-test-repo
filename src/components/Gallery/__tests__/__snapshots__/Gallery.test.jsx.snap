// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Gallery> should render correctly 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 Wrapper-n67dtg-0 eOOIQd"
    data-gtm-category="others"
    id="gallery-id"
  >
    <div
      class="Box-sc-1bsf784-0 jrMedw"
      display="flex"
      width="100%"
    >
      <button
        class="Box-sc-1bsf784-0 Column-t2p57o-0 sc-1k44dfv-0 jrnKmA"
        data-gtm-interaction-type="gallery - image filter"
        data-gtm-title="exterior"
      >
        <p
          class="sc-1k44dfv-0 sc-1k44dfv-0 hFTuyR"
          color="black"
          font-family="default"
          font-weight="regular"
        >
          Exterior
        </p>
      </button>
    </div>
    <div
      class="Box-sc-1bsf784-0 hyVGoO"
      height="100%"
    >
      <button
        aria-label="Shared.Gallery.fullscreenModeAria"
        class="IconWrapper-lp9q2r-0 sc-1k44dfv-0 hHFAcS"
      />
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 bYZXpx"
        height="180px,432px,702px"
        width="100%"
      >
        <div
          class="Box-sc-1bsf784-0 sc-1k44dfv-0 clIyAz"
        >
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 jazOpQ"
          >
            <div
              aria-live="polite"
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 jlohdu"
              height="100%"
              id="carousel-1"
            >
              <div
                class="Box-sc-1bsf784-0 sc-1k44dfv-0 jyKEsn"
              >
                <img
                  alt=""
                  class="sc-1k44dfv-0 cbPQHC"
                  data-gtm-category="others"
                  draggable="true"
                  height="1589"
                  loading="lazy"
                  src="http://via.placeholder.com/1248x702/8F8F8F?text=Placeholder_Image_1248x702_1"
                  width="2400"
                />
              </div>
              <div
                class="Box-sc-1bsf784-0 sc-1k44dfv-0 kwdYIQ"
              >
                <img
                  alt=""
                  class="sc-1k44dfv-0 fJDGLW"
                  data-gtm-category="others"
                  draggable="true"
                  height="1603"
                  loading="lazy"
                  src="http://via.placeholder.com/1248x702/808080?text=Placeholder_Image_1248x702_2"
                  width="2400"
                />
              </div>
              <div
                class="Box-sc-1bsf784-0 sc-1k44dfv-0 kwdYIQ"
              >
                <img
                  alt=""
                  class="sc-1k44dfv-0 fJDGLW"
                  data-gtm-category="others"
                  draggable="true"
                  height="1603"
                  loading="lazy"
                  src="http://via.placeholder.com/1248x702/8F8F8F?text=Placeholder_Image_1248x702_3"
                  width="2400"
                />
              </div>
              <div
                class="Box-sc-1bsf784-0 sc-1k44dfv-0 kwdYIQ"
              >
                <img
                  alt=""
                  class="sc-1k44dfv-0 hFRoqg"
                  data-gtm-category="others"
                  draggable="true"
                  height="344"
                  loading="lazy"
                  src="http://via.placeholder.com/1248x702/808080?text=Placeholder_Image_1248x702_4"
                  width="612"
                />
              </div>
              <div
                class="Box-sc-1bsf784-0 sc-1k44dfv-0 kwdYIQ"
              >
                <img
                  alt=""
                  class="sc-1k44dfv-0 hFRoqg"
                  data-gtm-category="others"
                  draggable="true"
                  height="344"
                  loading="lazy"
                  src="http://via.placeholder.com/1248x702/8F8F8F?text=Placeholder_Image_1248x702_5"
                  width="612"
                />
              </div>
              <div
                class="Box-sc-1bsf784-0 sc-1k44dfv-0 kwdYIQ"
              >
                <img
                  alt=""
                  class="sc-1k44dfv-0 bmxlN"
                  data-gtm-category="others"
                  draggable="true"
                  height="1310"
                  loading="lazy"
                  src="http://via.placeholder.com/1248x702/808080?text=Placeholder_Image_1248x702_6"
                  width="2400"
                />
              </div>
              <div
                class="Box-sc-1bsf784-0 sc-1k44dfv-0 kwdYIQ"
              >
                <img
                  alt=""
                  class="sc-1k44dfv-0 bmxlN"
                  data-gtm-category="others"
                  draggable="true"
                  height="1310"
                  loading="lazy"
                  src="http://via.placeholder.com/1248x702/8F8F8F?text=Placeholder_Image_1248x702_7"
                  width="2400"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="Box-sc-1bsf784-0 XgdIW"
          display="flex"
          height="100%"
        >
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 kijLGK"
            direction="in"
            id=""
          >
            <button
              aria-controls="carousel-1"
              aria-label="Shared.Gallery.prevImageAria"
              class="IconWrapper-lp9q2r-0 sc-bdVaJa jfRUiK"
              cursor="initial"
              data-gtm-interaction-type="gallery - view previous"
              data-gtm-title="exterior"
              disabled=""
            />
          </div>
        </div>
        <div
          class="Box-sc-1bsf784-0 exiPIC"
          display="flex"
          height="100%"
        >
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 kijLGK"
            direction="in"
            id=""
          >
            <button
              aria-controls="carousel-1"
              aria-label="Shared.Gallery.nextImageAria"
              class="IconWrapper-lp9q2r-0 sc-bdVaJa eIsOEp"
              cursor="pointer"
              data-gtm-interaction-type="gallery - view next"
              data-gtm-title="exterior"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="Box-sc-1bsf784-0 cLmRZY"
      display="flex"
      id="thumbnails-2"
      overflow="hidden"
      width="100%"
    >
      <button
        class="Box-sc-1bsf784-0 Column-t2p57o-0 sc-1k44dfv-0 iLLehK"
        width="0.25,0.25,0.16666666666666666"
      >
        <img
          alt=""
          class="sc-1k44dfv-0 cbPQHC"
          data-gtm-category="others"
          draggable="true"
          height="1589"
          loading="lazy"
          src="http://via.placeholder.com/1248x702/8F8F8F?text=Placeholder_Image_1248x702_1"
          width="2400"
        />
        <div
          class="Box-sc-1bsf784-0 QoWEs thumbnail-border"
          height="4px"
        />
      </button>
      <button
        class="Box-sc-1bsf784-0 Column-t2p57o-0 sc-1k44dfv-0 kyVUow"
        width="0.25,0.25,0.16666666666666666"
      >
        <img
          alt=""
          class="sc-1k44dfv-0 fJDGLW"
          data-gtm-category="others"
          draggable="true"
          height="1603"
          loading="lazy"
          src="http://via.placeholder.com/1248x702/808080?text=Placeholder_Image_1248x702_2"
          width="2400"
        />
        <div
          class="Box-sc-1bsf784-0 QoWEs thumbnail-border"
          height="4px"
        />
      </button>
      <button
        class="Box-sc-1bsf784-0 Column-t2p57o-0 sc-1k44dfv-0 kyVUow"
        width="0.25,0.25,0.16666666666666666"
      >
        <img
          alt=""
          class="sc-1k44dfv-0 fJDGLW"
          data-gtm-category="others"
          draggable="true"
          height="1603"
          loading="lazy"
          src="http://via.placeholder.com/1248x702/8F8F8F?text=Placeholder_Image_1248x702_3"
          width="2400"
        />
        <div
          class="Box-sc-1bsf784-0 QoWEs thumbnail-border"
          height="4px"
        />
      </button>
      <button
        class="Box-sc-1bsf784-0 Column-t2p57o-0 sc-1k44dfv-0 kyVUow"
        width="0.25,0.25,0.16666666666666666"
      >
        <img
          alt=""
          class="sc-1k44dfv-0 hFRoqg"
          data-gtm-category="others"
          draggable="true"
          height="344"
          loading="lazy"
          src="http://via.placeholder.com/1248x702/808080?text=Placeholder_Image_1248x702_4"
          width="612"
        />
        <div
          class="Box-sc-1bsf784-0 QoWEs thumbnail-border"
          height="4px"
        />
      </button>
      <button
        class="Box-sc-1bsf784-0 Column-t2p57o-0 sc-1k44dfv-0 kyVUow"
        width="0.25,0.25,0.16666666666666666"
      >
        <img
          alt=""
          class="sc-1k44dfv-0 hFRoqg"
          data-gtm-category="others"
          draggable="true"
          height="344"
          loading="lazy"
          src="http://via.placeholder.com/1248x702/8F8F8F?text=Placeholder_Image_1248x702_5"
          width="612"
        />
        <div
          class="Box-sc-1bsf784-0 QoWEs thumbnail-border"
          height="4px"
        />
      </button>
      <button
        class="Box-sc-1bsf784-0 Column-t2p57o-0 sc-1k44dfv-0 kyVUow"
        width="0.25,0.25,0.16666666666666666"
      >
        <img
          alt=""
          class="sc-1k44dfv-0 bmxlN"
          data-gtm-category="others"
          draggable="true"
          height="1310"
          loading="lazy"
          src="http://via.placeholder.com/1248x702/808080?text=Placeholder_Image_1248x702_6"
          width="2400"
        />
        <div
          class="Box-sc-1bsf784-0 QoWEs thumbnail-border"
          height="4px"
        />
      </button>
      <button
        class="Box-sc-1bsf784-0 Column-t2p57o-0 sc-1k44dfv-0 kyVUow"
        width="0.25,0.25,0.16666666666666666"
      >
        <img
          alt=""
          class="sc-1k44dfv-0 bmxlN"
          data-gtm-category="others"
          draggable="true"
          height="1310"
          loading="lazy"
          src="http://via.placeholder.com/1248x702/8F8F8F?text=Placeholder_Image_1248x702_7"
          width="2400"
        />
        <div
          class="Box-sc-1bsf784-0 QoWEs thumbnail-border"
          height="4px"
        />
      </button>
    </div>
    <div
      class="Box-sc-1bsf784-0 gKONMQ sc-1k44dfv-0 dlpEjh"
      display="flex"
      width="100%"
    >
      <button
        aria-controls="thumbnails-2"
        aria-label="Shared.Gallery.prevThumbnailPageAria"
        class="IconWrapper-lp9q2r-0 sc-bdVaJa jfRUiK"
        cursor="initial"
        disabled=""
      />
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 evNJbO"
      >
        <button
          aria-label="0"
          class="sc-1k44dfv-0 cjSExO"
          data-testid="step1"
          role="button"
        />
        <button
          aria-label="1"
          class="sc-1k44dfv-0 ipUiMI"
          data-testid="step2"
          role="button"
        />
      </div>
      <button
        aria-controls="thumbnails-2"
        aria-label="Shared.Gallery.nextThumbnailPageAria"
        class="IconWrapper-lp9q2r-0 sc-bdVaJa eIsOEp"
        cursor="pointer"
      />
    </div>
  </div>
</div>
`;

exports[`Gallery Short Variant should render short variant correctly 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 Wrapper-n67dtg-0 eOOIQd"
    data-gtm-category="others"
    id="gallery-id"
  >
    <div
      class="Box-sc-1bsf784-0 jrMedw"
      display="flex"
      width="100%"
    >
      <button
        class="Box-sc-1bsf784-0 Column-t2p57o-0 sc-1k44dfv-0 jrnKmA"
        data-gtm-interaction-type="gallery - image filter"
        data-gtm-title="exterior"
      >
        <p
          class="sc-1k44dfv-0 sc-1k44dfv-0 hFTuyR"
          color="black"
          font-family="default"
          font-weight="regular"
        >
          Exterior
        </p>
      </button>
    </div>
    <div
      class="Box-sc-1bsf784-0 igbPGr"
    >
      <button
        aria-label="Shared.Gallery.fullscreenModeAria"
        class="IconWrapper-lp9q2r-0 sc-1k44dfv-0 hHFAcS"
      />
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 jmMsw"
        height="180px,432px,702px"
        width="100%"
      >
        <div
          class="Box-sc-1bsf784-0 sc-1k44dfv-0 clIyAz"
        >
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 jazOpQ"
          >
            <div
              aria-live="polite"
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 jlohdu"
              height="100%"
              id="carousel-3"
            >
              <div
                class="Box-sc-1bsf784-0 sc-1k44dfv-0 jyKEsn"
              >
                <img
                  alt=""
                  class="sc-1k44dfv-0 cbPQHC"
                  data-gtm-category="others"
                  draggable="true"
                  height="1589"
                  loading="lazy"
                  src="http://via.placeholder.com/1248x702/8F8F8F?text=Placeholder_Image_1248x702_1"
                  width="2400"
                />
              </div>
              <div
                class="Box-sc-1bsf784-0 sc-1k44dfv-0 kwdYIQ"
              >
                <img
                  alt=""
                  class="sc-1k44dfv-0 fJDGLW"
                  data-gtm-category="others"
                  draggable="true"
                  height="1603"
                  loading="lazy"
                  src="http://via.placeholder.com/1248x702/808080?text=Placeholder_Image_1248x702_2"
                  width="2400"
                />
              </div>
              <div
                class="Box-sc-1bsf784-0 sc-1k44dfv-0 kwdYIQ"
              >
                <img
                  alt=""
                  class="sc-1k44dfv-0 fJDGLW"
                  data-gtm-category="others"
                  draggable="true"
                  height="1603"
                  loading="lazy"
                  src="http://via.placeholder.com/1248x702/8F8F8F?text=Placeholder_Image_1248x702_3"
                  width="2400"
                />
              </div>
              <div
                class="Box-sc-1bsf784-0 sc-1k44dfv-0 kwdYIQ"
              >
                <img
                  alt=""
                  class="sc-1k44dfv-0 hFRoqg"
                  data-gtm-category="others"
                  draggable="true"
                  height="344"
                  loading="lazy"
                  src="http://via.placeholder.com/1248x702/808080?text=Placeholder_Image_1248x702_4"
                  width="612"
                />
              </div>
              <div
                class="Box-sc-1bsf784-0 sc-1k44dfv-0 kwdYIQ"
              >
                <img
                  alt=""
                  class="sc-1k44dfv-0 hFRoqg"
                  data-gtm-category="others"
                  draggable="true"
                  height="344"
                  loading="lazy"
                  src="http://via.placeholder.com/1248x702/8F8F8F?text=Placeholder_Image_1248x702_5"
                  width="612"
                />
              </div>
              <div
                class="Box-sc-1bsf784-0 sc-1k44dfv-0 kwdYIQ"
              >
                <img
                  alt=""
                  class="sc-1k44dfv-0 bmxlN"
                  data-gtm-category="others"
                  draggable="true"
                  height="1310"
                  loading="lazy"
                  src="http://via.placeholder.com/1248x702/808080?text=Placeholder_Image_1248x702_6"
                  width="2400"
                />
              </div>
              <div
                class="Box-sc-1bsf784-0 sc-1k44dfv-0 kwdYIQ"
              >
                <img
                  alt=""
                  class="sc-1k44dfv-0 bmxlN"
                  data-gtm-category="others"
                  draggable="true"
                  height="1310"
                  loading="lazy"
                  src="http://via.placeholder.com/1248x702/8F8F8F?text=Placeholder_Image_1248x702_7"
                  width="2400"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="Box-sc-1bsf784-0 XgdIW"
          display="flex"
          height="100%"
        >
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 kijLGK"
            direction="in"
            id=""
          >
            <button
              aria-controls="carousel-3"
              aria-label="Shared.Gallery.prevImageAria"
              class="IconWrapper-lp9q2r-0 sc-bdVaJa jfRUiK"
              cursor="initial"
              data-gtm-interaction-type="gallery - view previous"
              data-gtm-title="exterior"
              disabled=""
            />
          </div>
        </div>
        <div
          class="Box-sc-1bsf784-0 exiPIC"
          display="flex"
          height="100%"
        >
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 kijLGK"
            direction="in"
            id=""
          >
            <button
              aria-controls="carousel-3"
              aria-label="Shared.Gallery.nextImageAria"
              class="IconWrapper-lp9q2r-0 sc-bdVaJa eIsOEp"
              cursor="pointer"
              data-gtm-interaction-type="gallery - view next"
              data-gtm-title="exterior"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="Box-sc-1bsf784-0 cLmRZY"
      display="flex"
      id="thumbnails-2"
      overflow="hidden"
      width="100%"
    >
      <button
        class="Box-sc-1bsf784-0 Column-t2p57o-0 sc-1k44dfv-0 iLLehK"
        width="0.25,0.25,0.16666666666666666"
      >
        <img
          alt=""
          class="sc-1k44dfv-0 cbPQHC"
          data-gtm-category="others"
          draggable="true"
          height="1589"
          loading="lazy"
          src="http://via.placeholder.com/1248x702/8F8F8F?text=Placeholder_Image_1248x702_1"
          width="2400"
        />
        <div
          class="Box-sc-1bsf784-0 QoWEs thumbnail-border"
          height="4px"
        />
      </button>
      <button
        class="Box-sc-1bsf784-0 Column-t2p57o-0 sc-1k44dfv-0 kyVUow"
        width="0.25,0.25,0.16666666666666666"
      >
        <img
          alt=""
          class="sc-1k44dfv-0 fJDGLW"
          data-gtm-category="others"
          draggable="true"
          height="1603"
          loading="lazy"
          src="http://via.placeholder.com/1248x702/808080?text=Placeholder_Image_1248x702_2"
          width="2400"
        />
        <div
          class="Box-sc-1bsf784-0 QoWEs thumbnail-border"
          height="4px"
        />
      </button>
      <button
        class="Box-sc-1bsf784-0 Column-t2p57o-0 sc-1k44dfv-0 kyVUow"
        width="0.25,0.25,0.16666666666666666"
      >
        <img
          alt=""
          class="sc-1k44dfv-0 fJDGLW"
          data-gtm-category="others"
          draggable="true"
          height="1603"
          loading="lazy"
          src="http://via.placeholder.com/1248x702/8F8F8F?text=Placeholder_Image_1248x702_3"
          width="2400"
        />
        <div
          class="Box-sc-1bsf784-0 QoWEs thumbnail-border"
          height="4px"
        />
      </button>
      <button
        class="Box-sc-1bsf784-0 Column-t2p57o-0 sc-1k44dfv-0 kyVUow"
        width="0.25,0.25,0.16666666666666666"
      >
        <img
          alt=""
          class="sc-1k44dfv-0 hFRoqg"
          data-gtm-category="others"
          draggable="true"
          height="344"
          loading="lazy"
          src="http://via.placeholder.com/1248x702/808080?text=Placeholder_Image_1248x702_4"
          width="612"
        />
        <div
          class="Box-sc-1bsf784-0 QoWEs thumbnail-border"
          height="4px"
        />
      </button>
      <button
        class="Box-sc-1bsf784-0 Column-t2p57o-0 sc-1k44dfv-0 kyVUow"
        width="0.25,0.25,0.16666666666666666"
      >
        <img
          alt=""
          class="sc-1k44dfv-0 hFRoqg"
          data-gtm-category="others"
          draggable="true"
          height="344"
          loading="lazy"
          src="http://via.placeholder.com/1248x702/8F8F8F?text=Placeholder_Image_1248x702_5"
          width="612"
        />
        <div
          class="Box-sc-1bsf784-0 QoWEs thumbnail-border"
          height="4px"
        />
      </button>
      <button
        class="Box-sc-1bsf784-0 Column-t2p57o-0 sc-1k44dfv-0 kyVUow"
        width="0.25,0.25,0.16666666666666666"
      >
        <img
          alt=""
          class="sc-1k44dfv-0 bmxlN"
          data-gtm-category="others"
          draggable="true"
          height="1310"
          loading="lazy"
          src="http://via.placeholder.com/1248x702/808080?text=Placeholder_Image_1248x702_6"
          width="2400"
        />
        <div
          class="Box-sc-1bsf784-0 QoWEs thumbnail-border"
          height="4px"
        />
      </button>
      <button
        class="Box-sc-1bsf784-0 Column-t2p57o-0 sc-1k44dfv-0 kyVUow"
        width="0.25,0.25,0.16666666666666666"
      >
        <img
          alt=""
          class="sc-1k44dfv-0 bmxlN"
          data-gtm-category="others"
          draggable="true"
          height="1310"
          loading="lazy"
          src="http://via.placeholder.com/1248x702/8F8F8F?text=Placeholder_Image_1248x702_7"
          width="2400"
        />
        <div
          class="Box-sc-1bsf784-0 QoWEs thumbnail-border"
          height="4px"
        />
      </button>
    </div>
    <div
      class="Box-sc-1bsf784-0 gKONMQ sc-1k44dfv-0 dlpEjh"
      display="flex"
      width="100%"
    >
      <button
        aria-controls="thumbnails-2"
        aria-label="Shared.Gallery.prevThumbnailPageAria"
        class="IconWrapper-lp9q2r-0 sc-bdVaJa jfRUiK"
        cursor="initial"
        disabled=""
      />
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 evNJbO"
      >
        <button
          aria-label="0"
          class="sc-1k44dfv-0 cjSExO"
          data-testid="step1"
          role="button"
        />
        <button
          aria-label="1"
          class="sc-1k44dfv-0 ipUiMI"
          data-testid="step2"
          role="button"
        />
      </div>
      <button
        aria-controls="thumbnails-2"
        aria-label="Shared.Gallery.nextThumbnailPageAria"
        class="IconWrapper-lp9q2r-0 sc-bdVaJa eIsOEp"
        cursor="pointer"
      />
    </div>
  </div>
</div>
`;
